### ISSUE TITLE
Ensure that Voice-to text users can activate links and inputs

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#6954]
### Description
Eric to go through the site ensuring the alternative nav (Voice-to text users for example) can navigate elements (buttons, links & inputs). Search and fix all.

> I noticed that your apply now link in the bottom middle did not have a little number next to it like it should with the show numbers command. I just wanted to bring this to your attention, so it could get corrected in your efforts to make your website accessible."

**Comment by esizer** [comment]
I have completed a full review of the applicant facing application (basically everything except for paths prefixed with `/admin`). I was able to reproduce **_all_** issues with enough effort. I believe it comes down to 2 issues. One is a minor annoyance we should try and fix and the other is absolutely frustrating and should probably be fixed as soon as we have time. 😖 

## Item Numbers not Appearing

This one, while annoying, is not a breaking issue. Though, I would still say we look into this since it can be breaking for new or inexperienced voice control users.

### The Issue

I believe this has to do with our loading strategy and is more apparent on slower connections. What happens when a user navigates to a form/new page,

1. We load the "page" as fast as possible, <- _this is where voice control inserts the item number overlays_
2. We then make a call to the API to gather information for a page
3. The page is then updated with that information <- _including new controls that should have item number overlays_

What is believe the issue is, we have some controls that depends on data we receive from the server. Voice control seems to only add numbers once the page loads and does not regenerate them once the page loads.

If a user simply re-iterates "show numbers" this forces the numbers to be regenerated and then they pick up the new controls. This is happens less frequently on faster connections since the server data can be loaded before the initial generation of numbers.

https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/95909808-e014-441b-b52d-1623445f7a1e

### The Fix

This is going to be difficult the fix and I'm not sure, at the moment, we have a good solution. The preferred solution would be to retrieve all data we have for a page before rendering it but this would be a fundamental change from our current strategy. 😬 

## Location Multiselect

This one is funadmentally broken and should be fixed as soon as possible.

### The Issue

I believe this is all mutliselect fields we currently have and not just the one on the search page. This input swallows all input and traps the user, it also re-implements the basic browser functionality and is nearly impossible to interact in any expected way with voice control because of this.

While the input contains instructions, those instructions are only announced to screen-readers so voice control users are left unaware of how to interact with the input so it becomes a guessing game. As you can see in the attached video, I'm kind of grasping at straws trying to invoke the options like a normal select so I can see the list of options and outright fail and give up 😭 

> **Note**
> This does not represent all of the iterations I tested. It was the last in my final attempt that ended up getting recorded

https://github.com/GCTC-NTGC/gc-digital-talent/assets/4127998/727c2e0e-265b-432e-8f77-6702fecb0380

### The Fix

I think the easiest fix would be to replace this with a list of checkboxes until we have time to research and test a multiselect combox component.

**Comment by petertgiles** [comment]
Regarding the "Numbers not appearing" issue, could this be partially resolved by using React Router loaders, where the bulk of the data for the page is retrieved before navigation completes?

Regarding the "Multiselect" issue, I like the short-term solution recommended.  I guess the long-term solution is to push forward with #5507 ?

**Comment by esizer** [comment]
> Regarding the "Numbers not appearing" issue, could this be partially resolved by using React Router loaders, where the bulk of the data for the page is retrieved before navigation completes?

Yeah, that was my thought. We could test it out with a page that is known to have this issue (create pool) and see if it resolves that. This would mean not using hooks with urql though 😬 They have a [guide on how to use data libraries with loaders](https://reactrouter.com/en/main/guides/data-libs) and I was able to get it working.

> Regarding the "Multiselect" issue, I like the short-term solution recommended. I guess the long-term solution is to push forward with #5507 ?

That is the long-term goal. We already have some issues with this input and I think this is just another nail in the coffin 💀 I think we just need confirmation from either @GCTC-NTGC/designers or @gobyrne  if we can replace that input on the search page with a checklist for now.

**Comment by Jerryescandon** [comment]
Yeah, I think a checklist as a temporary (or permanent) replacement for the **work locations** on the search page is an acceptable solution. Dont let it hold back #5507
